### PR TITLE
Re-implement scuba gear for 1.14

### PR DESCRIPTION
**Pull request by tterrag1098**
- Remove dive computer
- Merge into harness
- Make goggles improve visibility underwater

- Remove tanks
- Harness comes with its own tank(s)
- Compressor accepts harness

- Pony bottle now increases air over time instead of instantly

- Remove BCD/Weight Belt

- No recipes yet
- No trimix tier yet

### CODE REVIEW

File: src/main/java/net/tropicraft/core/client/scuba/ModelScubaGear.java [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+        mouthpiece2.showModel = showChest;
+        mouthpiece3.showModel = showChest;
+        hose1.showModel = showChest;
+        hose2.showModel = showChest;
+        hose3.showModel = showChest;
+        hose4.showModel = showChest;
+        hose5.showModel = showChest;
+        hose6.showModel = showChest;
+
+        // TODO what was this doing?
```
Comment (Cojomax99) [inline src/main/java/net/tropicraft/core/client/scuba/ModelScubaGear.java:435]: There's an air hose connecting the backpack to the face mask, I believe that's what this was.

File: src/main/java/net/tropicraft/core/client/scuba/ScubaAmbienceTicker.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    @SubscribeEvent
+    public static void onClientTick(ClientTickEvent event) {
+        if (event.phase != Phase.START) return;
+        Minecraft mc = Minecraft.getInstance();
+        if (mc.world != null && mc.player != null) {
+            ActiveRenderInfo renderInfo = mc.getRenderManager().info;
+            Entity renderViewEntity = mc.getRenderViewEntity();
+            if (renderInfo != null && renderViewEntity instanceof PlayerEntity) {
+                PlayerEntity player = (PlayerEntity) renderViewEntity;
+                if (renderInfo.getFluidState().isTagged(FluidTags.WATER) && player.getItemStackFromSlot(EquipmentSlotType.CHEST).getItem() instanceof ScubaArmorItem) {
```
Comment (Cojomax99) [inline src/main/java/net/tropicraft/core/client/scuba/ScubaAmbienceTicker.java:41]: nit: we could use item tags here, not a necessary change though.

File: src/main/java/net/tropicraft/core/client/scuba/ScubaHUD.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                    airColor = TextFormatting.GOLD;
+                } else {
+                    airColor = TextFormatting.GREEN;
+                }
+                double depth = ScubaData.getDepth(player);
+                String depthStr;
+                if (depth > 0) {
+                    depthStr = String.format("%.1fm", depth); 
+                } else {
+                    depthStr = "N/A"; // TODO translate
```
Comment (Cojomax99) [inline src/main/java/net/tropicraft/core/client/scuba/ScubaHUD.java]: TODO now or later?

File: src/main/java/net/tropicraft/core/common/block/tileentity/AirCompressorTileEntity.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+		this.compressing = false;
+		this.ticks = 0;
+		syncInventory();
+	}
+	
+	public float getBreatheProgress(float partialTicks) {
+	    if (isDoneCompressing()) {
+	        return 0;
+	    }
+	    return (float) (((((ticks + partialTicks) / 20) * Math.PI) + Math.PI) % (Math.PI * 2));
```
Comment (Cojomax99) [inline src/main/java/net/tropicraft/core/common/block/tileentity/AirCompressorTileEntity.java:154]: so much pi

File: src/main/java/net/tropicraft/core/common/dimension/TropicraftDimension.java [hunk 5]
```diff
@@ -123,6 +123,11 @@ public boolean isSurfaceWorld() {
     public float getCloudHeight() {
         return 192;
     }
+    
+    @Override
+    public int getSeaLevel() {
+        return 127; // TODO don't hardcode this both here and TropicraftChunkGenerator
```
Comment (Cojomax99) [inline src/main/java/net/tropicraft/core/common/dimension/TropicraftDimension.java:129]: move to Constants?

File: src/main/java/net/tropicraft/core/client/scuba/ScubaAmbienceTicker.java (hunk 2 above)
Comment (tterrag1098) [reply src/main/java/net/tropicraft/core/client/scuba/ScubaAmbienceTicker.java:41]: I'm not in love with tags that imply hierarchy, users could easily break the code

File: src/main/java/net/tropicraft/core/common/dimension/TropicraftDimension.java (hunk 5 above)
Comment (tterrag1098) [reply src/main/java/net/tropicraft/core/common/dimension/TropicraftDimension.java:129]: I'll do that on the main branch, it's out of scope here. The whole thing needs a refactor to defer to the Dimension always.

File: src/main/java/net/tropicraft/core/client/scuba/ModelScubaGear.java (hunk 1 above)
Comment (tterrag1098) [reply src/main/java/net/tropicraft/core/client/scuba/ModelScubaGear.java:435]: It didn't look right at all. I think it looks okay without it, and debugging the model rotations would take a while.

**Comment by Cojomax99** [comment]
image
Could be nice to show how much air remains in it in the tooltip. A nice-to-have, but a useful nice-to-have

File: src/main/java/net/tropicraft/core/common/item/scuba/ScubaHarnessItem.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    public boolean providesAir() {
+        return true;
+    }
+
+    @Override
+    public void tickAir(ServerPlayerEntity player, EquipmentSlotType slot, ItemStack stack) {
+        CompoundNBT scubaData = stack.getOrCreateChildTag("scuba");
+        int remaining = getRemainingAir(stack);
+        if (remaining > 0) {
+            scubaData.putInt("air", remaining - 1);
```
Comment (Cojomax99) [inline src/main/java/net/tropicraft/core/common/item/scuba/ScubaHarnessItem.java:51]: Air shouldn't go down if player is in creative mode

**Comment by Cojomax99** [comment]
Idea for the future: We should totally add a "Fast Flippers" enchantment to go faster while scubaing

**Comment by Cojomax99** [comment]
Missing recipes?

**Review by Cojomax99 — CHANGES_REQUESTED** [review]
I've gameplay tested, and it's super fun! I see a lot of potential with this. My only request is to add recipes. Once it has recipes, it has my approval.

**Review by Cojomax99 — APPROVED** [review]

**Comment by Cojomax99** [comment]
Approving, with the caveat that we need to add recipes for this 🧣

**Pull request merged**